test/format/import.c

```diff
@@ -0,0 +1,212 @@
+            cursor->set_value(cursor, counter + i);
+            testutil_check(cursor->insert(cursor));
+        }
+        counter += IMPORT_ENTRIES;
+        testutil_check(import_session->checkpoint(import_session, NULL));
+
+        /* Copy table into current test/format directory */
+        copy_file_into_directory(session, IMPORT_DIR, "import.wt");
+
+        import_value = mmrand(NULL, 0, 1);
+        if (import_value == 0) {
+            import_with_repair(session);
+        } else {
+            import_with_file_metadata(session, import_session);
+        }
+
+        /* Drop import table, so we can import next iteration */
+        while ((ret = session->drop(session, IMPORT_URI, NULL)) == EBUSY) {
+            __wt_yield();
+        }
+        testutil_check(ret);
+
+        verify_import(import_session, counter - IMPORT_ENTRIES);
+        while (period > 0 && !g.workers_finished) {
+            --period;
+            __wt_sleep(1, 0);
+        }
+    }
+
+    testutil_check(cursor->close(cursor));
+    wts_close(&import_conn, &import_session);
+    testutil_check(session->close(session, NULL));
+    return (WT_THREAD_RET_VALUE);
+}
+
+/*
+ * verify_import --
+ *     Verify all the values inside the imported table.
+ */
+static void
+verify_import(WT_SESSION *session, int start_value)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    int counter, key, value;
+
+    counter = 0;
+    testutil_check(session->open_cursor(session, IMPORT_URI, NULL, NULL, &cursor));
+
+    while ((ret = cursor->next(cursor)) == 0) {
+        error_check(cursor->get_key(cursor, &key));
+        testutil_assert(key == counter);
+        error_check(cursor->get_value(cursor, &value));
+        testutil_assert(value == counter + start_value);
+        counter++;
+    }
+    testutil_assert(counter == IMPORT_ENTRIES);
+    scan_end_check(ret == WT_NOTFOUND);
+}
+
+/*
+ * import_with_repair --
+ *     Perform import with repair option
+ */
+static void
+import_with_repair(WT_SESSION *session)
+{
+    WT_DECL_RET;
+    char buf[256];
+
+    memset(buf, 0, sizeof(buf));
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "import=(enabled,repair=true)"));
+    if ((ret = session->create(session, IMPORT_URI, buf)) != 0)
+        testutil_die(ret, "session.import", ret);
+}
+
+/*
+ * import_with_file_metadata --
+ *     Perform import with the file metadata.
+ */
+static void
+import_with_file_metadata(WT_SESSION *session, WT_SESSION *import_session)
+{
+    WT_CURSOR *metadata_cursor;
+    WT_DECL_RET;
+    char buf[2048];
+    const char *table_config, *file_config;
+
+    memset(buf, 0, sizeof(buf));
+    testutil_check(
+      import_session->open_cursor(import_session, "metadata:", NULL, NULL, &metadata_cursor));
+
+    metadata_cursor->set_key(metadata_cursor, IMPORT_URI);
+    metadata_cursor->search(metadata_cursor);
+    metadata_cursor->get_value(metadata_cursor, &table_config);
+
+    metadata_cursor->set_key(metadata_cursor, "file:import.wt");
```

Comment:
I would make `file:import.wt` a `#define IMPORT_URI_FILE` so that it is up with `IMPORT_URI` and someone would know to modify both in tandem. (And rename `IMPORT_URI_CONFIG` so that the two table names are next to each other.)

Reply:
done!